lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -636,19 +636,28 @@ namespace Js
                 }
                 else
                 {
-                    isAmbiguous = !childModuleRecord->ResolveExport(importNameId, resolveSet, exportRecord);
-                    if (isAmbiguous)
+                    if (exportEntry.importName->GetPropertyId() == PropertyIds::star_)
                     {
-                        // ambiguous; don't need to search further
+                        // export * as someName from "foo"
+                        *exportRecord = childModuleRecord->GetNamespaceNameRecord();
                         return true;
```

Comment:
I think it may be easier to understand control flow here if the else conditions were eliminated (I know they were here already and you were following the same style). Since each if ends in a return, we can eliminate them altogether.